### ISSUE TITLE
CNTK backend - Recurrent Models with sequences of mixed length 

### ISSUE THREAD

**Issue by smsaladi** [keras-team/keras#10017]
#40 suggests padding sequences with 0s and/or partitioning sequences into variable-size batches. At the time, CNTK was not available. 

Is it possible to take advantage of CNTK's native support for variable length sequences in keras?

https://docs.microsoft.com/en-us/python/cognitive-toolkit/sequence?view=cntk-py-2.5

If this question is better suited for SO, please let me know. I'll close and move it there. In case it's helpful, I've found the following potentially related issues: #7055 #7983 #7125 #2375

**Comment by tbennun** [comment]
@smsaladi I hope this will be helpful, but I had the same problem and just wrote a generic component within Keras, extending the `utils.Sequence` dataset reader: https://github.com/tbennun/keras-bucketed-sequence

Hope this helps, let me know if that works with CNTK. (I only tested this with Tensorflow)

**Comment by smsaladi** [comment]
Thanks -- I actually ended up doing something almost the same, also with the tf backend
